pkg/status/taskrunpod.go

```diff
@@ -55,8 +56,16 @@ func UpdateStatusFromPod(taskRun *v1alpha1.TaskRun, pod *corev1.Pod, resourceLis
 				ContainerName:  s.Name,
 				ImageID:        s.ImageID,
 			})
+		} else {
+			sidecars = append(taskRun.Status.Sidecars, v1alpha1.SidecarState{
+				Name:    s.Name,
+				ImageID: s.ImageID,
+			})
 		}
 	}
+	if len(sidecars) > 0 {
```

Comment:
You could just get rid of the `sidecars` slice and on line 60 do `taskRun.Status.Sidecars = append(taskRun.Status.Sidecars, v1alpha1.SidecarState{...})` like we do for step containers.

Reply:
I did that first and didn't want to update all the tests, but I switched to be consistent.